crates/nu-command/src/strings/str_/trim/command.rs

```diff
@@ -0,0 +1,1160 @@
+                Some('r'),
+            )
+            .switch(
+                "all",
+                "trims all characters from both sides of the string *and* in the middle (default: whitespace)",
+                Some('a'),
+            )
+            .switch("both", "trims all characters from left and right side of the string (default: whitespace)", Some('b'))
+            .switch("format", "trims spaces replacing multiple characters with singles in the middle (default: whitespace)", Some('f'))
+    }
+    fn usage(&self) -> &str {
+        "trims text"
+    }
+
+    fn run(
+        &self,
+        engine_state: &EngineState,
+        stack: &mut Stack,
+        call: &Call,
+        input: PipelineData,
+    ) -> Result<nu_protocol::PipelineData, nu_protocol::ShellError> {
+        operate(engine_state, stack, call, input, &trim)
+    }
+
+    fn examples(&self) -> Vec<Example> {
+        vec![
+            Example {
+                description: "Trim whitespace",
+                example: "'Nu shell ' | str trim",
+                result: Some(Value::test_string("Nu shell")),
+            },
+            Example {
+                description: "Trim a specific character",
+                example: "'=== Nu shell ===' | str trim -c '=' | str trim",
+                result: Some(Value::test_string("Nu shell")),
+            },
+            Example {
+                description: "Trim all characters",
+                example: "' Nu   shell ' | str trim -a",
+                result: Some(Value::test_string("Nushell")),
+            },
+            Example {
+                description: "Trim whitespace from the beginning of string",
+                example: "' Nu shell ' | str trim -l",
+                result: Some(Value::test_string("Nu shell ")),
+            },
+            Example {
+                description: "Trim a specific character",
+                example: "'=== Nu shell ===' | str trim -c '='",
+                result: Some(Value::test_string(" Nu shell ")),
+            },
+            Example {
+                description: "Trim whitespace from the end of string",
+                example: "' Nu shell ' | str trim -r",
+                result: Some(Value::test_string(" Nu shell")),
+            },
+            Example {
+                description: "Trim a specific character",
+                example: "'=== Nu shell ===' | str trim -r -c '='",
+                result: Some(Value::test_string("=== Nu shell ")),
+            },
+        ]
+    }
+}
+
+pub fn operate<F>(
+    engine_state: &EngineState,
+    stack: &mut Stack,
+    call: &Call,
+    input: PipelineData,
+    trim_operation: &'static F,
+) -> Result<nu_protocol::PipelineData, nu_protocol::ShellError>
+where
+    F: Fn(&str, Option<char>, &ClosureFlags) -> String + Send + Sync + 'static,
+{
+    let head = call.head;
+    let (options, closure_flags, input) = (
+        Arc::new(Arguments {
```

Comment:
You may be able to remove the Arc and just let the iterator move `Arguments` here

Reply:
Good catch, I missed that one :) I will fix it in another PR as this one was already merged. Thanks!